NorthstarDLL/plugins/plugins.cpp

```diff
@@ -1,421 +1,290 @@
-	serverInfo.name = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 2);
-	serverInfo.password = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 3);
-	gameState.players = g_pSquirrel<ScriptContext::UI>->getinteger(sqvm, 4);
-	serverInfo.maxPlayers = g_pSquirrel<ScriptContext::UI>->getinteger(sqvm, 5);
-	gameState.map = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 6);
-	gameState.mapDisplayName = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 7);
-	gameState.playlist = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 8);
-	gameState.playlistDisplayName = g_pSquirrel<ScriptContext::UI>->getstring(sqvm, 9);
-	ReleaseSRWLockExclusive(&gameStateLock);
-	ReleaseSRWLockExclusive(&serverInfoLock);
-	return SQRESULT_NOTNULL;
+	switch (type)
+	{
+	case ObjectType::CONVAR:
+		return (void*)new ConVar;
+	case ObjectType::CONCOMMANDS:
+		return (void*)new ConCommand;
+	}
 }
 
-// int maxPlayers
-SQRESULT SQ_UpdateServerInfoBetweenRounds(HSquirrelVM* sqvm)
+std::optional<Plugin> PluginManager::LoadPlugin(fs::path path, PluginInitFuncs* funcs, PluginNorthstarData* data)
 {
-	AcquireSRWLockExclusive(&serverInfoLock);
-	serverInfo.id = g_pSquirrel<ScriptContext::CLIENT>->getstring(sqvm, 1);
-	serverInfo.name = g_pSquirrel<ScriptContext::CLIENT>->getstring(sqvm, 2);
-	serverInfo.password = g_pSquirrel<ScriptContext::CLIENT>->getstring(sqvm, 3);
-	serverInfo.maxPlayers = g_pSquirrel<ScriptContext::CLIENT>->getinteger(sqvm, 4);
-	ReleaseSRWLockExclusive(&serverInfoLock);
-	return SQRESULT_NOTNULL;
-}
 
-// float timeInFuture
-SQRESULT SQ_UpdateTimeInfo(HSquirrelVM* sqvm)
-{
-	AcquireSRWLockExclusive(&serverInfoLock);
-	serverInfo.endTime = ceil(g_pSquirrel<ScriptContext::CLIENT>->getfloat(sqvm, 1));
-	ReleaseSRWLockExclusive(&serverInfoLock);
-	return SQRESULT_NOTNULL;
-}
+	Plugin plugin {};
 
-// bool loading
-SQRESULT SQ_SetConnected(HSquirrelVM* sqvm)
-{
-	AcquireSRWLockExclusive(&gameStateLock);
-	gameState.loading = g_pSquirrel<ScriptContext::UI>->getbool(sqvm, 1);
-	ReleaseSRWLockExclusive(&gameStateLock);
-	return SQRESULT_NOTNULL;
-}
+	std::string pathstring = (pluginPath / path).string();
+	std::wstring wpath = (pluginPath / path).wstring();
 
-SQRESULT SQ_UpdateListenServer(HSquirrelVM* sqvm)
-{
-	AcquireSRWLockExclusive(&serverInfoLock);
-	serverInfo.id = g_pMasterServerManager->m_sOwnServerId;
-	serverInfo.password = ""; // g_pServerPresence->Cvar_ns_server_password->GetString(); todo this fr
-	ReleaseSRWLockExclusive(&serverInfoLock);
-	return SQRESULT_NOTNULL;
-}
+	LPCWSTR wpptr = wpath.c_str();
+	HMODULE datafile = LoadLibraryExW(wpptr, 0, LOAD_LIBRARY_AS_DATAFILE | LOAD_LIBRARY_AS_IMAGE_RESOURCE); // Load the DLL as a data file
+	if (datafile == NULL)
+	{
+		NS::log::PLUGINSYS->info("Failed to load library '{}': ", std::system_category().message(GetLastError()));
+		return std::nullopt;
+	}
+	HRSRC manifestResource = FindResourceW(datafile, MAKEINTRESOURCEW(101), MAKEINTRESOURCEW(RT_RCDATA));
```

Comment:
I know this constant also existed before this PR, but the magic number `101` really needs to be documented under some name.

In the NorthstarDiscordRPC rewrite its called `IDR_RCDATA1`
https://github.com/R2Northstar/NorthstarDiscordRPC/blob/52207395f269721a667f676a458a2864844a0a58/DiscordRPCV2/resource.h#L793

This is the name used to store the manifest under

Reply:
Addressed in 603dfc72a109d2043e456271abf3f33d259eb2c0